src/api/selection.js

```diff
@@ -10,10 +10,26 @@ function (elementHelper) {
      * Wrapper for object holding currently selected text.
      */
     function Selection() {
-      this.selection = window.getSelection();
+      var rootDoc = document;
 
+      // find the parent document or document fragment
+      var currentElement = scribe.el.parentNode;
+      while(currentElement && currentElement.nodeType !== 11 && currentElement.nodeType !== 9) { // 11 = DOCUMENT_FRAGMENT_NODE, 9 = DOCUMENT
```

Comment:
Please use constants on `Node` rather than ids that then need to be explicited by comments.

For example `Node.DOCUMENT_FRAGMENT_NODE` is `11`, etc.